protocol/messenger.go

```diff
@@ -2716,7 +2716,96 @@ func (m *Messenger) SyncDevices(ctx context.Context, ensName, photoPath string)
 		return err
 	}
 
-	return err
+	accounts, err := m.settings.GetAccounts()
+	if err != nil {
+		return err
+	}
+	return m.syncWallets(accounts)
+}
+
+// Sync wallets in case of account changes
+func (m *Messenger) watchAccountListChanges() {
+	m.logger.Debug("watching account changes")
+
+	channel := m.settings.SubscribeToAccountChanges()
+
+	go func() {
+		for {
+			select {
+			case accounts := <-channel:
+				err := m.syncWallets(accounts)
```

Comment:
Do we need to check if we've seen this account before? I think that wallet api passes a lot of things through this feed, I could be mistaken, but I've seen that the wallet API makes these kind of checks.

Reply:
Thanks, this is interesting. So this is invoked only from `SaveAccounts`/`DeleteAccounts`, its only purpose is send a list of changed accounts to other paired devices. We might repeatedly see the same account if it e.g. was deleted and then re-added again. What negative consequences might arise from this?